docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1021 @@
+Logstash 1.5, all plugins are self-contained ruby gems.  In all previous
+versions, plugins were part of the core Logstash distribution.  This change
+makes it possible to develop plugins independent of the Logstash core, and
+release updates independently as well!
+
+NOTE: Logstash is written in Ruby.  If you are unfamiliar with the Ruby programming
+language, an excellent quick-start guide is at
+https://www.ruby-lang.org/en/documentation/quickstart/.
+
+== Get started
+
+{getstarted}
+
+=== Create a new GitHub repo
+As of version 1.5 of Logstash, the code for each plugin lives in its own GitHub
+repository. So first, you'll need to create a new repo as a home for your
+nascent {plugintype}. When logged into GitHub, click on the "Repositories" tab.
+You may see a list of repositories you've forked and/or contributed to, or it
+may be blank. Either way, you're fine! Now click on the green "New" button near
+the top right of the page.
+
+Choose the following settings on your new repo:
+
+* Repository name should be similar to "logstash-{plugintype}-mypluginname"
+* Public or Private - up to you (but will need to be Public to submit as an
+  official plugin)
+* Click the box "Initialize this repository with a README"
+
+=== Copy the {plugintype} code
+
+==== Clone your forked repo
+You'll need to clone the new repo you just created. Use a command like this:
+
+[source,sh]
+[subs="attributes"]
+----------------------------------
+git clone git@github.com:myuser/logstash-{plugintype}-mypluginname.git
+----------------------------------
+
+==== Clone and copy the example plugin (highly recommended)
+At this point, you could build the plugin out entirely on your own.  Plugins
+need to have a similar directory structure as other ruby gems.
+
+[NOTE]
+=========================
+This example ruby gem file structure
+(and excellent walkthrough of ruby gem creation) is from
+http://timelessrepo.com/making-ruby-gems :
+[source,txt]
+[subs="attributes"]
+----------------------------------
+$ tree mygem
+├── lib/
+│   ├── mygem.rb
+│   └── mygem/
+│       ├── base.rb
+│       └── version.rb
+├── test/
+├── bin/
+├── Rakefile
+└── mygem.gemspec
+----------------------------------
+=========================
+
+For Logstash plugins, your file structure should look like this:
+
+[source,txt]
+[subs="attributes"]
+----------------------------------
+$ tree logstash-{plugintype}-mypluginname
+├── Gemfile
+├── LICENSE
+├── README.md
+├── Rakefile
+├── lib
+│   └── logstash
+│       └── {plugintype}s
+│           └── mypluginname.rb
+├── logstash-{plugintype}-mypluginname.gemspec
+└── spec
+    └── {plugintype}s
+        └── mypluginname_spec.rb
+----------------------------------
+
+
+[TIP]
+=========================
+Here's a relatively easy way to import the necessary files and
+directory structure.
+
+[source,js]
+[subs="attributes"]
+----------------------------------
+cd /tmp
+git clone git@github.com:logstash-plugins/logstash-{plugintype}-{pluginname}.git
```

Comment:
This doesn't work unless you have ssh set up.  This may not be obvious to the first time user (i.e. it wasn't obvious to me)  Perhaps refer users to https://help.github.com/articles/generating-ssh-keys/